### PR TITLE
shortcut '1' to '9' to league

### PR DESCRIPTION
**Pull request by LuqDragon**
add numbers key to enter in league gyms, I put 1 to 9 cause we can apply to others if have more than 5.
trying to make more generic now.

### CODE REVIEW

File: src/scripts/GameController.ts [hunk 1]
```diff
@@ -157,6 +157,11 @@ class GameController {
                         DungeonRunner.initializeDungeon(player.town().dungeon());
                     }
                     e.preventDefault();
+                } else if ('gymList' in player.town()) {
+                    const cap = keyCode > 48 && keyCode < 58 ? 48 : keyCode > 96 && keyCode < 106 ? 96 : -1; //1 to 9
+                    if (keyCode > cap && keyCode - cap <= player.town().gymList().length) {
+                        GymRunner.startGym(player.town().gymList()[keyCode - cap - 1]());
```
Comment (RedSparr0w) [inline src/scripts/GameController.ts:163]: I'm not sure if it's just me, but the logic seems a little bit confusing,
Maybe we could simplify it a little bit with something like:
```suggestion
                    const number = keyCode > 48 && keyCode < 58 ? keyCode - 48 : keyCode > 96 && keyCode < 106 ? keyCode - 96 : 0; //1 to 9
                    if (number && number <= player.town().gymList().length) {
                        GymRunner.startGym(player.town().gymList()[number - 1]());
```
or something like this:
```suggestion
                    const numberKey = keyCode > 48 && keyCode < 58 ? keyCode - 48 : 0;
                    const numpadKey = keyCode > 96 && keyCode < 106 ? keyCode - 96 : 0;
                    const number = numberKey || numpadKey;
                    if (number && number <= player.town().gymList().length) {
                        GymRunner.startGym(player.town().gymList()[number - 1]());
```

Comment (LuqDragon) [reply src/scripts/GameController.ts:163]: i liked the first one, but the second is more understandable

Comment (RedSparr0w) [reply src/scripts/GameController.ts:163]: I also just noticed #780 was merged recently, so this will probably need to be modified.

Comment (SpenserJ) [reply src/scripts/GameController.ts:154]: My vote is on the second one. I've never been a big fan of nested ternaries, especially when they're all on one line.

**Review by RedSparr0w — APPROVED** [review]
Working great 👍 

Thanks for the PR

**Pull request merged**